### PR TITLE
Implements separate ApplicationJobs for generating different AbIDs for those linked to ArchivesSpace records and those linked to MARC records

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #96 with the following:
- Implementing ApplicationJob Classes for separately generating AbsoluteIds for ArchivesSpace and MARC source records
- Restructuring the AbsoluteId Model in order to ensure that MARC-only AbsoluteIds can be generated

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
d537410, 918d4c8, and 325ca8b were necessary to restore the tests to passing on CircleCI. I can rebase this, or consider this blocked until #92 is merged (my preference is for the latter).

**Comment by jrgriffiniii** [comment]
Blocked by https://github.com/pulibrary/lib_jobs/pull/92

**Comment by jrgriffiniii** [comment]
Many of the bugs which were fixed within https://github.com/pulibrary/lib_jobs/tree/absolute-ids-top-containers-refactor seem to have been introduced in this PR. I am fixing these now.

File: app/jobs/absolute_id_create_record_job.rb
Comment (tpendragon) [inline app/jobs/absolute_id_create_record_job.rb:2]: This class is untested: https://1347-165745983-gh.circle-artifacts.com/0/coverage/index.html#d46fab0128696ee76db1970302d5be0a0905ae06

File: app/jobs/absolute_id_create_marc_source_record_job.rb
Comment (tpendragon) [inline app/jobs/absolute_id_create_marc_source_record_job.rb:93]: current_user/current_client are uncovered:

https://1347-165745983-gh.circle-artifacts.com/0/coverage/index.html#063449fe3044a43f7ce1474652937a6bd6110d32

**Review by tpendragon — CHANGES_REQUESTED** [review]

**Review by tpendragon — APPROVED** [review]

**Pull request merged**
